x/mongo/driver/topology/hanging_tls_conn_1_17.go

```diff
@@ -33,6 +33,12 @@ func newHangingTLSConn(conn *tls.Conn, sleepTime time.Duration) *hangingTLSConn
 
 // HandshakeContext implements the tlsConn interface on Go 1.17 and higher.
 func (h *hangingTLSConn) HandshakeContext(ctx context.Context) error {
-	time.Sleep(h.sleepTime)
+	timer := time.NewTimer(h.sleepTime)
```

Comment:
Nice fix. I suppose instead of a context leak this is a "timer leak"? Odd that I didn't see this locally on Go 1.17.3 🤔

Reply:
Yeah, it basically didn't return when requested by the `context.Context` timing out, and kept blocking (sleeping) instead.